src/library_webgl.js

```diff
@@ -2427,11 +2427,13 @@ var LibraryGL = {
     if (4*count <= GL.MINI_TEMP_BUFFER_SIZE) {
       // avoid allocation when uploading few enough uniforms
       var view = GL.miniTempBufferFloatViews[4*count-1];
+      // hoist the heap out of the loop for size and for pthreads+growth.
+      var heap = HEAPF32;
       for (var i = 0; i < 4*count; i += 4) {
-        view[i] = {{{ makeGetValue('value', '4*i', 'float') }}};
-        view[i+1] = {{{ makeGetValue('value', '4*i+4', 'float') }}};
-        view[i+2] = {{{ makeGetValue('value', '4*i+8', 'float') }}};
-        view[i+3] = {{{ makeGetValue('value', '4*i+12', 'float') }}};
+        view[i] = heap[value + 4*i >> 2];
+        view[i+1] = heap[value + 4*i+4 >> 2];
+        view[i+2] = heap[value + 4*i+8 >> 2];
+        view[i+3] = heap[value + 4*i+12 >> 2];
```

Comment:
Looks good - perhaps hoist the arithmetic as well?
```js
var heap = HEAPF32;
value >>= 2;
for (var end = value + count; value < end; value += 4) {
  view[i] = heap[value];
  view[i+1] = heap[value + 1];
  view[i+2] = heap[value + 2];
  view[i+3] = heap[value + 3];
}
```
?

Reply:
Oh, good point, I was worried that might change the semantics (if the pointer is unaligned) but all the added factors are just multiples of 4 so it's fine. Will do as a followup.